### ISSUE TITLE
EarlyStopping mode-parameter "auto" is missleading

### ISSUE THREAD

**Issue by Lennart4711** [keras-team/keras#17593]
In the [docs on EarlyStopping](https://keras.io/api/callbacks/early_stopping/) it says: "in 'auto' mode, the direction is automatically inferred from the name of the monitored quantity." suggesting there is some sort of "smart" way/ lookup table for different metrics. When you look at the  [corresponding code](https://github.com/keras-team/keras/blob/e6784e4302c7b8cd116b74a784f4b78d60e83c26/keras/callbacks.py#L2038) however, you see that it is just a string-comparision wether the name of the monitored variable ends in acc, accuracy or auc. 
```python
if (
    self.monitor.endswith("acc")
    or self.monitor.endswith("accuracy")
    or self.monitor.endswith("auc")
):
    self.monitor_op = np.greater
else:
    self.monitor_op = np.less
```
I believe this is missleading and causes confusion. 
You can see someone struggeling with it in [this stackoverflow post](https://stackoverflow.com/questions/69744405/tensorflow-earlystopping-stops-too-early) who thinks that "TF has the direction of ndcg (Normalized Discounted Cumulative Gain) for ranking problems wrong." when in reality it does not have direction for anything.

I don't know the consequences of changing this, as someone might have choosen the right value by accident and changing the standard value could break their pipeline(s). I do **not** have a fix because I don't know how widespread this problem is. I just wanted to let you know that this could be an issue worth looking into.

**Comment by Lennart4711** [comment]
This notation is used in this repo a couple of times, so I would suggest that.
```python
def __init__(mode='min'):
    if mode not in ["min", "max"]:
        raise ValueError(
            '`mode` must be one of `"min"`, `"max"`. '
            f"Received: mode={mode}."
        )
    self.monitor_op = np.greater if mode="max" else np.less

**Comment by tilakrayal** [comment]
@Lennart4711,
Could you please feel free to submit a PR for the requested change or share the link where requested change is to be made. Thank you!

**Comment by Lennart4711** [comment]
So in [this implementation](https://github.com/Lennart4711/keras/blob/e0fd29f6193b035f77da2b794f0d07972b7ad403/keras/callbacks.py#L2006) are the changes **iff** you want to remove the "auto"-mode. This would give a warning to everyone that used "auto"-mode previously. It would break the code for those who used "auto"-mode in combination with a monitored-quanitiy with a name ending in "acc", "accuracy" or "auc".

An alternativ option is to just change the parameter-description to something, that resembles the actual functionality better without changing the code at all. Something like this is actually already in use [here](https://github.com/keras-team/keras/blob/e6784e4302c7b8cd116b74a784f4b78d60e83c26/keras/callbacks.py#L1307) and I changed it [here](https://github.com/Lennart4711/keras/blob/c3d1d1592b045dedc96088387aa285dd73a999a7/keras/callbacks.py#L1972). 

I would advise to choose the latter option because it is less invasive and helps those who have similar problems to the guy in the linked stackoverflow above. If you agree, I am going to submit the PR for the latter.

**Comment by Lennart4711** [comment]
Same problem in one other class, [ReduceLROnPlateu](https://github.com/keras-team/keras/blob/e6784e4302c7b8cd116b74a784f4b78d60e83c26/keras/callbacks.py#L2906). Could be fixed/ changed the same as EarlyStopping.

**Comment by rchao** [comment]
Agreeing that taking 'auto' to automatically infer the direction is less clear than ideal, and being explicit in the argument's documentation without breaking the backward compatibility seems a better move. Happy to review a PR for the doc change.

**Comment by tilakrayal** [comment]
Hello, Thank you for reporting an issue.

We're currently in the process of migrating the new `Keras 3` code base from **keras-team/keras-core** to **keras-team/keras.**
Consequently, This issue may not be relevant to the `Keras 3 `code base. After the migration is successfully completed, feel free to reopen this issue at **keras-team/keras** if you believe it remains relevant to the `Keras 3` code base. 
If instead this issue is a bug or security issue in legacy **tf.keras**, you can instead report a new issue at [keras-team/tf-keras](https://github.com/keras-team/tf-keras/issues), which hosts the TensorFlow-only, legacy version of Keras.

To know more about `Keras 3,` please take a look at https://keras.io/keras_core/announcement/. Thank you!
